src/frontend/web_application/src/components/MessageList/components/MessageActionsContainer/index.jsx

```diff
@@ -1,88 +1,49 @@
     onMessageUnread: PropTypes.func.isRequired,
     onMessageRead: PropTypes.func.isRequired,
     __: PropTypes.func.isRequired,
   };
 
-  state = {
-    isRead: false,
-  };
-
-  componentWillMount() {
-    this.setState(generateStateFromProps(this.props));
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.setState(generateStateFromProps(nextProps));
-  }
-
   handleDelete = () => {
     const { message, onDelete } = this.props;
     onDelete({ message });
   };
 
-  handleToggle = (event) => {
-    const checked = event.target.checked;
+  handleToggle = () => {
     const { message, onMessageRead, onMessageUnread } = this.props;
 
-    if (checked) { onMessageRead({ message }); }
-    if (!checked) { onMessageUnread({ message }); }
-
-    this.setState(prevState => ({
-      isRead: checked,
-      ...prevState,
-    }));
+    if (message.is_unread) { onMessageRead({ message }); }
+    if (!message.is_unread) { onMessageUnread({ message }); }
   }
 
   render() {
-    const { __, message, dropdownId } = this.props;
+    const { __, message } = this.props;
 
     return (
       <div className="m-message-actions-container">
-        <DropdownControl toggle={dropdownId} className="m-message__actions-switcher">
-          <Icon type="ellipsis-v" />
-        </DropdownControl>
-        <Dropdown
-          id={dropdownId}
-          className="m-message__actions-menu"
-          position="left"
-          closeOnClick
-        >
-          <Button className="m-message-actions-container__action" icon="reply" responsive="icon-only">{__('message-list.message.action.reply')}</Button>
-          <Button className="m-message-actions-container__action" icon="share" responsive="icon-only">{__('message-list.message.action.copy-to')}</Button>
-          <CheckboxFieldGroup
-            className="m-message-actions-container__action"
-            label={message.is_unread ? __('message-list.message.status.unread') : __('message-list.message.status.read')}
-            name="isRead"
-            checked={!message.is_unread}
-            onChange={this.handleToggle}
-          />
-
-          <Button onClick={this.handleDelete} className="m-message-actions-container__action" icon="trash" responsive="icon-only">{__('message-list.message.action.delete')}</Button>
-        </Dropdown>
+        <Button className="m-message-actions-container__action" display="expanded" icon="reply" responsive="icon-only">{__('message-list.message.action.reply')}</Button>
+        <Button
+          className="m-message-actions-container__action"
+          onClick={this.handleDelete}
+          icon="trash"
+          responsive="icon-only"
+          display="expanded"
+        >{__('message-list.message.action.delete')}</Button>
+
+        <Button
+          className="m-message-actions-container__action"
+          display="expanded"
+          onClick={this.handleToggle}
+        >{message.is_unread ? 'Mark read' : 'Mark unread'}</Button>
```

Comment:
There is no latency when you click on checkbox ? 

In that case using `state.isRead` to have a fast responding checkbox is ok to me, sorry if I wasn't clear.

Reply:
here is missing translation func.